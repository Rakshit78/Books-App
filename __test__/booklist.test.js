import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

import Boolist from '../components/Booklist';

describe('render Booklist component', () => {
  it('render component without error with snapshot', () => {
    const tree = render(<Boolist />).toJSON();
    expect(tree).toMatchSnapshot();
    screen.getByText('Search for Books');
  });
});
