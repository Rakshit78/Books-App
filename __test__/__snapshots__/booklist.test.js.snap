// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Booklist component render component without error with snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#101010",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 28,
          "fontWeight": "700",
          "marginHorizontal": 28,
          "marginTop": 10,
        }
      }
    >
      Search for Books
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1A1A1A",
          "borderRadius": 30,
          "color": "#FFF",
          "flexDirection": "row",
          "height": 50,
          "marginHorizontal": "auto",
          "marginTop": 35,
          "padding": 10,
          "width": "90%",
        }
      }
    >
      <Text />
      <TextInput
        onChangeText={[Function]}
        placeholder="Search"
        placeholderTextColor="#C4B4A0"
        style={
          {
            "color": "#FFF",
            "fontSize": 18,
            "height": 30,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginHorizontal": "auto",
          "marginTop": -30,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "space-evenly",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 80,
        }
      }
    />
  </View>
</RCTScrollView>
`;
